cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -136,13 +136,15 @@ private static Future<PlatformFeaturesAvailability> createPlatformFeaturesAvaila
      *
      * @return  Future which completes when all Cluster Operator verticles are started and running
      */
-    static CompositeFuture deployClusterOperatorVerticles(Vertx vertx, KubernetesClient client, MetricsProvider metricsProvider, PlatformFeaturesAvailability pfa, ClusterOperatorConfig config, ShutdownHook shutdownHook) {
+    static CompositeFuture deployClusterOperatorVerticles(Vertx vertx, KubernetesClient client, MetricsProvider metricsProvider,
+                                                          PlatformFeaturesAvailability pfa, ClusterOperatorConfig config, ShutdownHook shutdownHook) {
         ResourceOperatorSupplier resourceOperatorSupplier = new ResourceOperatorSupplier(
                 vertx,
                 client,
                 metricsProvider,
                 pfa,
-                config.getOperatorName()
+                config.getOperatorName(),
+                config.featureGates().serverSideApplyEnabled()
```

Comment:
genuine question ... is it needed to pass the specific feature gates instead of passing all of them via `config.featureGates()` ? I mean, if meantime we add a new feature gate we would need to add an additional parameter here while we already have them in the collection?

Reply:
Yeah I guess I can pass the feature gates only there and the use the `serverSideApplyEnabled()` in the constructor where we are creating particular operators.

Reply:
I think I decided to not have the feature gates included there, as I would have to create empty config (so using `new FeatureGates("")` in one of the constructors - which seems to be used only in tests.